scripts/objectstore/copy_files_to_irods.py

```diff
@@ -0,0 +1,112 @@
+import os
+import subprocess
+import time
+
+from psycopg2 import connect, sql
+
+from galaxy.util import directory_hash_id
+
+
+def copy_files_to_irods(start_dataset_id, end_dataset_id, galaxy_dir="/Users/kxk302/workspace/galaxy", object_store_info={"files1" : "database/files1", "files2" : "database/files2"}, irods_home="/tempZone/home/rods", connection_info={"dbname" : "galaxy", "user" : "postgres", "host" : "localhost", "password" : "password"}):
+    conn = None
+
+    if start_dataset_id >= end_dataset_id:
+        print("start_dataset_id %d must be smaller than end_dataset_id %d", start_dataset_id, end_dataset_id)
+        return
+
+    try:
+        # declare a new PostgreSQL connection object
+        conn = connect(
+            dbname=connection_info["dbname"],
+            user=connection_info["user"],
+            host=connection_info["host"],
+            password=connection_info["password"]
+        )
+        conn.cursor()
+
+    except Exception as e:
+        print(e)
+        return
+
+    for dataset_id in range(start_dataset_id, end_dataset_id):
+        copy_file_to_irods(dataset_id, galaxy_dir, object_store_info, irods_home, conn)
+
+    conn.close()
+
+
+def copy_file_to_irods(dataset_id, galaxy_dir, object_store_info, irods_home, conn):
+    cursor = None
+    sql_statement = None
+    sql_object = None
+    uuid = None
+    uuid_with_dash = None
+    objectid = None
+    object_store_id = None
+    disk_sub_folder = None
+    disk_file_path = None
+    irods_sub_folder = None
+    irods_file_path = None
+    icommand_mkdir = None
+    icommand_put = None
+
+    sql_statement = "SELECT * FROM dataset WHERE state = \'ok\' AND deleted = False AND purged = False AND id = {};".format(dataset_id)
+    print(sql_statement)
+    try:
+        cursor = conn.cursor()
+
+        sql_object = sql.SQL(
+            sql_statement
+        )
+        cursor.execute(sql_object)
+        print("cursor.execute() completed")
+
+        table_data = cursor.fetchall()
+        print("cursor.fetchall() completed")
+        for num, row in enumerate(table_data):
+            print("row: ", row)
+            print("\n")
+            uuid = row[12]
```

Comment:
Column ordering isn't fixed on instances that ran schema migrations over time, [psycopg2.extras.namedtuple](https://www.psycopg.org/docs/extras.html#namedtuple-cursor) is a nice way to deal with this, but I would also explicitly list the columns you need in the `SELECT` rather than `SELECT *`.

Reply:
Sure

Reply:
I think if i specify the columns that I need in the SELECT statement, I should be good here.